### PR TITLE
Add MessageAttributes for SNS

### PR DESCRIPTION
**Pull request by Heanthor**
Thanks for this library, not having to do auth saved me a big headache 😅 .

The implementation of message attributes seemed to be incomplete in SNS, this is what I was able to make work for me. It does not include the binary type, but I tested strings, numbers, and string arrays. Plus supplying the type information doesn't seem necessary, we can determine it all.

### CODE REVIEW

**Comment by jcberquist** [comment]
Thanks! I can see that I just completely left out support for message attributes - appreciate the pull request.

File: services/sns.cfc [hunk 1]
```diff
@@ -139,6 +137,29 @@ component {
...
+                if ( isNumeric(v) ) {
+                    dataType = "Number";
+                } else if ( isArray(v) ) {
+                    dataType = "String.Array";
+                }
+
+                formParams [ "MessageAttributes.entry.#i#.Name" ] = k;
+                formParams [ "MessageAttributes.entry.#i#.Value.DataType" ] = dataType;
+                if ( dataType == "String.Array" ) {
+                    formParams [ "MessageAttributes.entry.#i#.Value.StringValue" ] = "[\""#v.toList("\"", \""")#\""]";
```
Comment (jcberquist) [inline services/sns.cfc]: Using `\"` to escape quotes is not a CFML thing. 😄 Also, I took a look at the docs and it looks like `String.Array` can include strings, numbers, and the keywords `true`, `false`, and `null`. Of those, only the strings need to be quoted. So I actually think this line could just use `serializeJSON(v)` to format the array correctly.

**Comment by jcberquist** [comment]
I would also like to see the `Binary` type supported for message attributes. I rewrote your addition slightly to look as follows:
```cfc
if ( arguments.keyExists( 'MessageAttributes' ) ) {
        var i = 1;
        MessageAttributes.each( function( k, v ) {
            var dataType = 'String';
            var valueType = 'StringValue';

            if ( isBinary( v ) ) {
                dataType = 'Binary';
                valueType = 'BinaryValue';
                v = binaryEncode( v, 'base64' );
            } else if ( isArray( v ) ) {
                dataType = 'String.Array';
                v = serializeJSON( v );
            } else if ( isNumeric( v ) ) {
                dataType = 'Number';
            }

            formParams[ 'MessageAttributes.entry.#i#.Name' ] = k;
            formParams[ 'MessageAttributes.entry.#i#.Value.DataType' ] = dataType;
            formParams[ 'MessageAttributes.entry.#i#.Value.#valueType#' ] = v;

            i++;
        } );
    }
```
This way, binary attributes are supported as well. What do you think?

**Comment by Heanthor** [comment]
Thanks, this looks improved. I must admit I only added the portion of the SNS API that I needed for my application, but I should have completed it before submitting this, my apologies! I have added your suggestions.

File: services/sns.cfc (hunk 1 above)
Comment (Heanthor) [reply services/sns.cfc]: I was directly copying one of the AWS Java examples for this portion, definitely not idiomatic coldfusion 😬 thanks

**Comment by jcberquist** [comment]
Thank you very much!

**Pull request merged**